pkg/updatestatus/nodeinformer.go

```diff
@@ -194,6 +202,37 @@ func (c *nodeInformerController) sync(ctx context.Context, syncCtx factory.SyncC
 	return nil
 }
 
+func (c *nodeInformerController) initializeCachesNoErrors() {
+	if err := c.initializeCaches(); err != nil {
+		klog.Errorf("Failed to initialize caches: %v", err)
+	}
+}
+
+func (c *nodeInformerController) initializeCaches() error {
+	var errs []error
+
+	if pools, err := c.machineConfigPools.List(labels.Everything()); err != nil {
+		errs = append(errs, err)
+	} else {
+		for _, pool := range pools {
+			c.machineConfigPoolSelectorCache.ingest(pool)
+		}
+	}
+	klog.V(2).Infof("Stored %d machineConfigPools in the cache", c.machineConfigPoolSelectorCache.len())
+
+	machineConfigs, err := c.machineConfigs.List(labels.Everything())
+	if err != nil {
+		errs = append(errs, err)
+	} else {
+		for _, mc := range machineConfigs {
+			c.machineConfigVersionCache.ingest(mc)
+		}
+	}
+	klog.V(2).Infof("Stored %d machineConfig versions in the cache", c.machineConfigVersionCache.len())
+
+	return kutilerrors.NewAggregate(errs)
+}
```

Comment:
I'd prefer to not couple the two into a single method: they are similar but are used in a different way at different places - consider having separate dedicated methods (which saves us from `NewAggregate` so the two methods would even be simpler, and have two `Once` instances.

Reply:
Two init methods now but still they are called in the same Once instance.

https://github.com/openshift/cluster-version-operator/pull/1164/commits/a81ec69e31f504ce10a44650cfa5c4d4260fd2cd#diff-c5a41cb83200e13544e0509a5e77c180e16a1d70d7411a79cac2bc107e89950fR99